tests/console/console.go

```diff
@@ -312,7 +314,30 @@ func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batch
 		}
 	}
 
+	// Detect the first `BatchSend` and inject before an expectation for the prompt.
```

Comment:
We don't just need this before the first send, we need it before every send, as you described in https://github.com/kubevirt/kubevirt/pull/14474#issuecomment-2858870655

The easiest is probably to create a new Batcher slice instead of altering the one passed to the function, that way the loop above can simply add 3 items (send \n, expect prompt, send origBatch[i]) to the new slice for every `BatchSend`.

Reply:
So we have 4 options now:
1. What was initially proposed in https://github.com/kubevirt/kubevirt/pull/14474
2. What was proposed in this PR.
3. Extend the logic in this PR and inject the prompt expecter before any `BSnd` (perhaps optimizing to the cases when one already exists).
4. Go ahead and introduce a new abstraction [1] and change all the codebase.

We could go with 3 and start preparing 4, moving things gradually to it while declaring the old ones deprecated to discourage usage.

[1] https://github.com/kubevirt/kubevirt/pull/14474#issuecomment-3211304560